src/ban.cpp

```diff
@@ -44,15 +44,15 @@ bool Ban::acceptConnection(const Connection::Address& clientIP)
 	return true;
 }
 
-bool IOBan::isAccountBanned(uint32_t accountId, BanInfo& banInfo)
+const std::shared_ptr<BanInfo> IOBan::isAccountBanned(uint32_t accountId)
```

Comment:
`std::optional<BanInfo>` is a better option, since it will be stack allocated and use less memory.